torch/_inductor/select_algorithm.py

```diff
@@ -789,11 +789,15 @@ def call_name(self):
         return f"template_kernels.{self.name}"
 
     def hash_key(self):
-        return "-".join(
-            [
-                self.name.rsplit("_", 1)[0],
-                self.bmreq.module_cache_key,
-            ]
+        return (
```

Comment:
Hmm, would it make more sense to log the correspondence of full key and key as log entry ?  It's could be a bit of a foot gun that being more verbose here causes local/global cache misses.

Reply:
What if we added a super clear "DO NOT USE THIS UNLESS YOU KNOW WHAT YOU ARE DOING" warning to the config? In my mind, this is meant for a very small subset of people (us) who want to do things like pruning/heuristic analysis/etc. for which they need the template arguments.